flagd/pkg/service/flag-evaluation/flag_evaluator_v2.go

```diff
@@ -107,15 +108,29 @@ func (s *FlagEvaluationService) ResolveAll(
 				s.logger.ErrorWithID(reqID, fmt.Sprintf("struct response construction: %v", err))
 				continue
 			}
-			res.Flags[value.FlagKey] = &evalV1.AnyFlag{
-				Reason:  value.Reason,
-				Variant: value.Variant,
+			res.Flags[resolved.FlagKey] = &evalV1.AnyFlag{
+				Reason:  resolved.Reason,
+				Variant: resolved.Variant,
 				Value: &evalV1.AnyFlag_ObjectValue{
 					ObjectValue: val,
 				},
 			}
 		}
+		// TODO: error handling
```

Comment:
Is there additional error handling required?

Reply:
No this is left over.